### PR TITLE
Include "select all" button for resource selector.

### PR DESCRIPTION
**Pull request by kimisgold**
This addresses #1695. The resource select sidebar now has a "select all" button. It behaves as follows:

* In its initial state, it is inactive and appears with an empty checkbox. Screen readers will read it as "Select all is inactive". 
  * When the user clicks on its initial state (unchecked), the "quick add" checkboxes open up, and all the checkboxes are ticked. Screen readers will read a success message that says "Quick add checkboxes are now available. All results are now selected."
  * The button itself then displays a ticked checkbox icon. When focused, screen readers will read it as "Select all is active."
* If the user clicks the active "select all" button and no individual resource has been unchecked, it will uncheck all boxes. The "select all" button returns to its initial inactive state.
* If the user deselects a resource while the "select all" button is active, it goes back to its inactive state.
  * Clicking the "select all" button again immediately after will again select all resources and return the button to its active state.
* If the user toggles off the "quick add" interface while all resources are selected, the "select all" button will remain in its active state, indicating that even if the checkboxes are not visible, they are checked.

### CODE REVIEW

File: application/view/omeka/admin/item/sidebar-select.phtml [hunk 1]
```diff
@@ -64,13 +64,18 @@ $expanded = $resourceClassId || $itemSetId || $id;
     </div>
 
     <?php if ($itemsFound): ?>
-    <button type="button" class="quick-select-toggle"><?php echo $translate('Quick add'); ?></button>
+    <span class="sr-only quick-select-on success" tabindex="-1"><?php echo $translate('Quick add checkboxes are now available.'); ?></span>
+    <span class="sr-only quick-select-off success" tabindex="-1"><?php echo $translate('Quick add checkboxes are now hidden.'); ?></span>
+    <button type="button" class="quick-select-toggle"><?php echo $translate('Quick add'); ?> <span class="sr-only active"><?php echo $translate('Is active'); ?></span><span class="sr-only inactive"><?php echo $translate('Inactive'); ?></span></button>
+    <span class="sr-only select-all-on success" tabindex="-1"><?php echo $translate(sprintf('%s All results are now selected.', $translate('Quick add checkboxes are now available.'))); ?></span>
```
Comment (zerocrates) [inline application/view/omeka/admin/item/sidebar-select.phtml:70]: Reversed order of the translate and sprintf calls here (the first or "outer" one specifically).

File: application/view/omeka/admin/item/sidebar-select.phtml [hunk 2]
```diff
@@ -64,13 +64,18 @@ $expanded = $resourceClassId || $itemSetId || $id;
     </div>
 
     <?php if ($itemsFound): ?>
-    <button type="button" class="quick-select-toggle"><?php echo $translate('Quick add'); ?></button>
+    <span class="sr-only quick-select-on success" tabindex="-1"><?php echo $translate('Quick add checkboxes are now available.'); ?></span>
```
Comment (zerocrates) [inline application/view/omeka/admin/item/sidebar-select.phtml:67]: It seems to me that these two spans are always screenreader-visible, am I right on that? Seems like they could get read outside of being explicitly focused by the JS.

File: application/asset/js/resource-selector.js [hunk 3]
```diff
@@ -52,8 +53,34 @@
         });
 
         $('#select-resource').on('click', '.quick-select-toggle', function() {
-            $(this).toggleClass('active');
+            $(this).toggleClass('active').find('button .sr-only').toggleClass('active');
```
Comment (zerocrates) [inline application/asset/js/resource-selector.js:56]: Little odd to toggle the "active" class on the screenreader text inside the button but not the "inactive" class... I don't quite get why this is necessary. You have a comment on the corresponding styles but I'm not getting the distinction.

File: application/view/omeka/admin/item/sidebar-select.phtml (hunk 2 above)
Comment (kimisgold) [reply application/view/omeka/admin/item/sidebar-select.phtml:67]: That's a good point, as I was only testing via tabbing through. They should probably have `display: none` when not active.

File: application/asset/js/resource-selector.js (hunk 3 above)
Comment (kimisgold) [reply application/asset/js/resource-selector.js:56]: I don't think it's necessary after all. I think these classes and their toggling got away from me, gonna pare them down.

**Comment by zerocrates** [comment]
My testing indicates this breaks the loading spinner for the sidebar: since this adds the "regular" weight fa icons, and the style for the loading spinner asks for regular weight. It really needs bold, but previously when we only had the bold weight ones, it worked. Fix is simply just asking for bold in that style, but worth a little search through to see if the same thing is being done elsewhere.

Separate problem: since this was created we merged in a feature to extend quickadd to the item set sidebar also, so this should be used there also.

**Pull request merged**